File: frontend/de/unika/ipd/grgen/util/GraphDumper.java
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 4.5
 * Copyright (C) 2003-2019 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3 (see LICENSE.txt included in the packaging of this file)
 * www.grgen.net
 */

package de.unika.ipd.grgen.util;

import java.awt.Color;

/**
 * A Dumper for Graphs
 */
public interface GraphDumper {

	int DEFAULT = -1;

	int BOX = 0;
	int RHOMB = 1;
	int ELLIPSE = 2;
	int TRIANGLE = 3;

	int SOLID = 0;
	int DASHED = 1;
	int DOTTED = 2;

	void begin();
	void finish();

	void beginSubgraph(GraphDumpable d);
	void beginSubgraph(String name);
	void endSubgraph();

	void node(GraphDumpable d);

	void edge(GraphDumpable from, GraphDumpable to, String label, int style,
						Color color);

	void edge(GraphDumpable from, GraphDumpable to, String label, int style);
	void edge(GraphDumpable from, GraphDumpable to, String label);
	void edge(GraphDumpable from, GraphDumpable to);
}

File: frontend/de/unika/ipd/grgen/util/GraphDumper.cs
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 8.1
 * Copyright (C) 2003-2026 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3, some components/parts use different licenses (see LICENSE.txt included in the packaging of this file)
 * www.grgen.de / www.grgen.net
 */

namespace de.unika.ipd.grgen.util
{

	// originally color definitions from java.awt.Color (including RGB values)
	public enum Color
	{
		BLACK, BLUE, CYAN, GRAY, DARK_GRAY, MAGENTA, ORANGE, GREEN, RED, PINK, YELLOW, WHITE
	}

	public enum Shape
	{
		DEFAULT = -1, BOX, RHOMB, ELLIPSE, TRIANGLE
	}

	public enum Style
	{
		DEFAULT = -1, SOLID, DASHED, DOTTED
	}

	/// <summary>
	/// A Dumper for Graphs
	/// </summary>
	public interface GraphDumper
	{
		void Begin();

		void Finish();

		void BeginSubgraph(GraphDumpable d);

		void BeginSubgraph(string name);

		void EndSubgraph();

		void Node(GraphDumpable d);

		void Edge(GraphDumpable from, GraphDumpable to, string label, Style style, Color color);

		void Edge(GraphDumpable from, GraphDumpable to, string label, Style style);

		void Edge(GraphDumpable from, GraphDumpable to, string label);

		void Edge(GraphDumpable from, GraphDumpable to);
	}

}
